data/scenarios/Challenges/Mazes/loopy_maze.yaml

```diff
@@ -0,0 +1,87 @@
+name: Invisible, loopy maze
+description: There is a maze, but it can't be seen, only sensed... can you
+  program a robot to navigate it successfully?
+objectives:
+- goal:
+  - There is a maze, but it can't be seen, only sensed... can you
+    program a robot to navigate it successfully?  You need to get a robot
+    to the goal square, labelled with an exclamation mark; you win by `grab`bing
+    the `goal`.
+  - In this challenge, you are NOT guaranteed that the maze is a tree, that is,
+    the maze may contain loops.
+  condition: |
+    try {
+      teleport self (27, -17);
+      b <- ishere "goal";
+      return (not b)
+    } { return false }
+solution: |
+  run "data/scenarios/Challenges/Mazes/loopy_maze_sol.sw"
+robots:
+  - name: base
+    dir: [0,1]
+    display:
+      char: 'Ω'
+      attr: robot
+    devices:
+      - dictionary
+      - 3D printer
+      - logger
+      - grabber
+    inventory:
+      - [50, solar panel]
+      - [50, treads]
+      - [50, compass]
+      - [50, scanner]
+      - [50, lambda]
+      - [50, branch predictor]
+      - [50, strange loop]
+      - [50, logger]
+      - [50, grabber]
+      - [1000, rock]
```

Comment:
A cool upgrade to the challenge would be adding a workbench and requiring the shortest path of e.g. furnaces from start to finish. :fire:

Reply:
Ah, that does sound like fun.  How would you do it, though?  I thought a little bit about how to do BFS and I am not sure how to do it.  Hmm, maybe a sort of iterative deepening thing, where we explore everywhere we can get in one step, then everywhere we can get in at most two steps, etc...